### ISSUE TITLE
SQL Server connector returns incorrect results for old `datetimeoffset` values

### ISSUE THREAD

**Issue by ebyhr** [trinodb/trino#16559]
Prepare the table on SQL Server
```
create table dbo.test (c1 DATETIMEOFFSET(3));
insert into dbo.test values ('2020-09-27 00:00:00.1234567+07:00');
insert into dbo.test values ('1400-09-27 00:00:00.1234567+07:00');
select * from dbo.test;
c1
---------------------------------------------
               2020-09-27 00:00:00.123 +07:00
               1400-09-27 00:00:00.123 +07:00
```

Select the table on Trino
```
trino:dbo> select * from dbo.test;
               c1
--------------------------------
 2020-09-27 00:00:00.123 +07:00
 1400-10-06 00:00:00.123 +07:00
```

**Comment by sudip-unb** [comment]
I am interested to work on this issue.

**Comment by ebyhr** [comment]
@sudip-unb Could you coordinate with @vlad-lyutenko?

**Comment by vlad-lyutenko** [comment]
Please go ahead, I didn't start it yet

**Comment by pauliev** [comment]
I am hitting this issue as well...

**Comment by mosabua** [comment]
Wasnt that fixed recently with the linked PR#19627  @ebyhr @vlad-lyutenko and hence #16559 could be closed.
